### ISSUE TITLE
I think your model suffers from overfitting for colors

### ISSUE THREAD

**Issue by 0xprincess** [kosuke1701/AnimeCV#1]
I used a notebook to experiment with your model, and what I found, that it's misidentify characters with similar color scheme, that are well-presented in a train data.
On the image: Sinon (x2), Earth-chan, rem with slightly changed color.
![image](https://user-images.githubusercontent.com/12896581/107155663-ac2d8600-698a-11eb-8783-d3a5e161c9d1.png)

**Comment by 0xprincess** [comment]
Maybe this issue can be solved by changing a threshold. But it also can make results worse for detection of the same character, so it needs to be changed based on data (model perfomance on test set, given the fixed threshold), not intuiton.

**Comment by 0xprincess** [comment]
The first model I tested on was 0111_best_randaug.
I tried 0206_resnet18 and that gives much better results, but there's some errors too, and a lot of results on the edge of a threshold. 
![image](https://user-images.githubusercontent.com/12896581/107156008-d5e7ac80-698c-11eb-8d0e-cc54b432dd95.png)

**Comment by 0xprincess** [comment]
0206_seresnet152 is not much better
![image](https://user-images.githubusercontent.com/12896581/107156061-25c67380-698d-11eb-9c59-abb5cff0c413.png)

0206_resnet152 gives the best result here
![image](https://user-images.githubusercontent.com/12896581/107156175-d6cd0e00-698d-11eb-908a-48af619a2e37.png)

**Comment by 0xprincess** [comment]
Added more samples with the same hair color. The results are bad even on 0206_resnet152
![image](https://user-images.githubusercontent.com/12896581/107157137-7e990a80-6993-11eb-866f-4c4ff39ec5c0.png)

**Comment by kosuke1701** [comment]
Thank you very much for detail inspection to my models!

As you pointed out, over-fitting to hair color is a very likely problem, and I think we have to consider different training strategies (negative sampling, data augmentation (e.g. using gray scaled image)) or inference method (one naive idea is to use both original and gray scaled image and aggregate these results but I'm not sure whether it works because the model is not trained for  such images).

> Maybe this issue can be solved by changing a threshold. But it also can make results worse for detection of the same character, so it needs to be changed based on data (model perfomance on test set, given the fixed threshold), not intuiton.

Yes, you are right. The threshold in the colab notebook is computed for 0111_best_randaug so that its false negative ratio would be the same to that of humans. I've computed thresholds by applying the same method to other models. (I'm sorry to have missed writing these values.) Interestingly, the adjusted thresholds are close to that of 0111_best_randaug, and thus unfortunately, using these new thresholds won't solve the hair-color problem.

* 0206_resnet18: 0.601
* 0206_resnet152: 0.645
* 0206_seresnet152: 0.656

**Comment by kosuke1701** [comment]
Hmm, to evaluate model's ability more correctly, I think we have to update test data so that it contains more confusing image pairs with the same hair color. I think adversarially sampling negative image pairs will do the job.
